doc/source/whatsnew/v0.25.0.rst

```diff
@@ -183,6 +183,7 @@ Reshaping
 - Bug in :func:`pandas.merge` adds a string of ``None`` if ``None`` is assigned in suffixes instead of remain the column name as-is (:issue:`24782`).
 - Bug in :func:`merge` when merging by index name would sometimes result in an incorrectly numbered index (:issue:`24212`)
 - :func:`to_records` now accepts dtypes to its `column_dtypes` parameter (:issue:`24895`)
+- Bug in :func:`concat` order of OrderedDict is not respected (:issue:`21510`)
```

Comment:
Bug in :func:`concat` where the order of a ``OrderedDict`` is not respected (use double backticks on OrderedDict)